receiver/libhoneyreceiver/factory.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package libhoneyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/libhoneyreceiver"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/libhoneyreceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/libhoneyreceiver/internal/sharedcomponent"
+)
+
+const (
+	httpPort = 8080
+)
+
+var defaultTracesURLPaths = []string{"/events", "/event", "/batch"}
+
+// NewFactory creates a new OTLP receiver factory.
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		metadata.Type,
+		createDefaultConfig,
+		receiver.WithTraces(createTraces, metadata.TracesStability),
+		receiver.WithLogs(createLogs, metadata.LogsStability),
+	)
+}
+
+// createDefaultConfig creates the default configuration for receiver.
+func createDefaultConfig() component.Config {
+	durationFieldsArr := []string{"duration_ms"}
+	endpointStr := fmt.Sprintf("localhost:%d", httpPort)
+	return &Config{
+		HTTP: &HTTPConfig{
+			ServerConfig: &confighttp.ServerConfig{
+				Endpoint: endpointStr,
+			},
+			TracesURLPaths: defaultTracesURLPaths,
+		},
+		AuthAPI: "",
+		Resources: ResourcesConfig{
+			ServiceName: "service.name",
```

Comment:
The point of these is to be able to map non-semconv service name strings into a the semconv named string.  

The use of `service.name` here as a default is just because people may have already changed their service name fields to match what OTEL uses and we want to honor that unless they want to configure an override.